engineering/documentation.md

```diff
@@ -0,0 +1,241 @@
+# Documentation at source{d}
+
+Our company has been a big believer on Open Source from very early on.
+Most of our repositories are visible to anyone in the world.
+We agree this is for our own benefit, but public source code is not
+enough to support and develop an Open Source community.
+
+The goal of this document is to help source{d} engineers understand
+what pieces of documentation they should write for an open source project,
+and how they should be written.
+
+The main audience of this document is source{d} engineers, but it might
+be a useful resource for many others inside and outside our company.
+
+This document is divided into two sections:
+
+1. The first section covers somes best practices for technical writing which
+apply to any piece of documentation you might be writing.
+1. The second section lists the main documentation artifacts all source{d}
+projects should provide.
+
+## General Best Practices
+
+People read documentation in order to achieve a goal.
+Whenever you write a new piece you should keep their goal in mind.
+
+For instance, you're currently reading this document.
+What is your goal?
+Maybe you want to improve the documentation of your own project,
+or perhaps you've been told this is an awesome document and your goal
+is simply to enjoy reading a beautiful text of astonishing beauty.
+
+On the other hand I know you're a developer, so there's no point explaining
+what source code is. That would seem patronizing or a waste of time.
+
+I also know that most of you don't speak English as a first language.
```

Comment:
`s/most of you/most of us/` ? 

Otherwise, together with `I`, might be perceived as a little bit.. well, alienating. 
What do you think?

Reply:
Good point! 😄
done